internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -63,3 +66,17 @@ func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 func (cd *ConnectionDetails) Address() string {
 	return cd.Host + ":" + strconv.FormatUint(uint64(cd.Port), 10)
 }
+
+func (cd *ConnectionDetails) parseSSLOptions(credentials map[string][]byte) {
+	cd.SSLOptions = make(map[string]string)
+
+	for _, sslOption := range sslOptions {
+		if len(credentials[sslOption]) > 0 {
+			value := string(credentials[sslOption])
+			if value != "" {
+				cd.SSLOptions[sslOption] = value
+			}
+		}
+		delete(credentials, sslOption)
+	}
```

Comment:
I think zeroization should take place in a defer at the top of NewConnectionDetails, so that we get all the key-value pairs. That way we don't need zeroization in multiple places (i.e. here and https://github.com/cyberark/secretless-broker/pull/1184)

Reply:
@doodlesbykumbi I don't think this is even zeroizing.  It's just removing the key.  You'd be relying on GC to remove it, even assuming it has no other refs.

Also, didn't we have a generic solution for zeroizing at the framework level above this?